webapp/client/client.jsx

```diff
@@ -22,13 +22,24 @@ export default class Client {
         this.defaultHeaders = {
             'X-Requested-With': 'XMLHttpRequest'
         };
+        this.redirect_url = null;
```

Comment:
Doing this in the Client feels wrong to me. It works since it's initialized statically, but this doesn't relate to the client at all.

It would be better to do something like this in the `preNeedsTeam` function in `route_team.jsx` when it first redirects the browser to the root route in the first place. You can also use the arguments provided by React Router there to avoid having to manually parse the redirect URL as well. If you change [this line](https://github.com/mattermost/platform/blob/master/webapp/routes/route_team.jsx#L97) to redirect to `'/?redirect_to=' + encodeURIComponent(nextState.location.pathname)`, you'll have the `redirect_to` parameter already set when you get to the Root component without needing any special handling there